sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java

```diff
@@ -694,7 +694,15 @@ public static Row toBeamRow(Schema rowSchema, TableSchema bqSchema, TableRow jso
     if (jsonBQValue instanceof List) {
       return ((List<Object>) jsonBQValue)
           .stream()
-              .map(v -> ((Map<String, Object>) v).get("v"))
+              .map(
+                  v ->
+                      (v instanceof Map
+                              && !fieldType
+                                  .getCollectionElementType()
```

Comment:
are their NPE concern? fieldType.getCollectionElementType is nullable

Reply:
and `!fieldType.getCollectionElementType().getTypeName().equals(TypeName.MAP)` does not depend on and can be extract outside the lambda (for visibility)

Reply:
`fieldType.getCollectionElementType()` should only return `null` if the fieldType is not a collection, but the `if (jsonBQValue instanceof List)` case should cover that scenario. However, there's nothing stopping someone from passing in a non-collection FieldType with a List value, so my latest changes show a more helpful error message in that case.

I've extracted the `!fieldType.getCollectionElementType().getTypeName().equals(TypeName.MAP)` to a variable outside of the lambda.